### PR TITLE
Allow forcing work of LDS_MaintainSimplifiedLayers no matter dataset

### LINKED ISSUE #154: Allow passing 0 to MaintainSimplifiedLayers and friends to mean force regeneration
Right now most `MaintainXXXLayers` functions accept a `job id` as a way to skip generating some derived tables IFF the given upload identifier did not change any of their source tables. This is not practical when willing to run those functions after changing their code or after a new upload that did not trigger calling those functions.

It also prevents running them outside of the upload transaction in that they query tables that are only available temporarely within the upload transaction, making these functions depend on `linz-bde-uploader` code too (see #94).

This ticket is to enhance these functions to accept a special `upload ID` value (I'm proposing `0`) to mean that they should not try to be smart and check source tables. Doing so should ideally allow running the function w/out even having any `bde_control` schema, and ensure all relative tables are built.

### PR DESCRIPTION
**Pull request by strk**
Closes #154

### CODE REVIEW

File: sql/04-lds_layer_functions.sql.in [hunk 1]
```diff
@@ -86,10 +86,16 @@ RETURNS
 DECLARE
     v_dataset        TEXT;
 BEGIN
-    v_dataset := COALESCE(
-        bde_control.bde_GetOption(p_upload_id, '_dataset'),
-        '(undefined dataset)'
-    );
+    IF p_upload_id != 0 THEN
+        v_dataset := COALESCE(
+            bde_control.bde_GetOption(p_upload_id, '_dataset'),
```
Comment (palmerj) [inline sql/04-lds_layer_functions.sql.in]: Don't think this function will work without the temp bde upload schema being created first

Comment (palmerj) [reply sql/04-lds_layer_functions.sql.in]: See here https://github.com/linz/linz-bde-uploader/blob/master/sql/02-bde_control_functions.sql.in#L814

Comment (strk) [reply sql/04-lds_layer_functions.sql.in]: The goal is NOT calling GetOptions at all (that part I didn't do yet)

Comment (palmerj) [reply sql/04-lds_layer_functions.sql.in]: Ah, long day. Yes you are correct.

Comment (palmerj) [reply sql/04-lds_layer_functions.sql.in]: I just reviewed all other bde_control.bde_xx calls and bde_GetOption is the only one that needs the schema so the other function calls deeper in this function will be ok. Good work

**Comment by strk** [comment]
With latest commit (rebased now) I did wrap the GetOption calls into the check for p_upload being != 0.
Things should work now.

TODO:
 -  [ ] Quick testcase (calling the function could be enough for a start?)
 -  [ ] CHANGELOG entry

**Pull request merged**